### PR TITLE
Update runtime spec and tools dependencies

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind dependency-change

#### What this PR does / why we need it:
We now stop inning the runtime-spec and update both dependencies to its latest commit.

#### Which issue(s) this PR fixes:

None
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by TomSweeneyRedHat** [comment]
LGTM
and happy green test buttons.

**Comment by haircommander** [comment]
/hold

@fidencio @fgiudici PTAL at kata failures

**Comment by fidencio** [comment]
The failure we're seeing as part of the kata-containers CI is:
```
21:51:18 Install CNI plugins
21:51:18 Retrieve CNI plugins repository
21:51:18 go: downloading github.com/containernetworking/plugins v0.9.1
21:51:20 .ci/install_cni_plugins.sh: line 16: pushd: /tmp/jenkins/workspace/kata-containers-2-crio-PR/go/src/github.com/containernetworking/plugins: No such file or directory
21:51:20 Failed at 16: pushd $GOPATH/src/github.com/containernetworking/plugins
21:51:20 Build step 'Execute shell' marked build as failure
```

The failure doesn't look related to CRI-O, at all.

@haircommander, for how long can we hold this PR?  I don't have cycles to take a look at this one Today.

**Comment by haircommander** [comment]
tomorrow is fine, go to bed :stuck_out_tongue:

**Comment by fidencio** [comment]
Actually, it may be a simple fix, let me stop the current jobs and give it a try.

**Comment by fidencio** [comment]
> The failure we're seeing as part of the kata-containers CI is:
> 
> ```
> 21:51:18 Install CNI plugins
> 21:51:18 Retrieve CNI plugins repository
> 21:51:18 go: downloading github.com/containernetworking/plugins v0.9.1
> 21:51:20 .ci/install_cni_plugins.sh: line 16: pushd: /tmp/jenkins/workspace/kata-containers-2-crio-PR/go/src/github.com/containernetworking/plugins: No such file or directory
> 21:51:20 Failed at 16: pushd $GOPATH/src/github.com/containernetworking/plugins
> 21:51:20 Build step 'Execute shell' marked build as failure
> ```
> 
> The failure doesn't look related to CRI-O, at all.
> 
> @haircommander, for how long can we hold this PR? I don't have cycles to take a look at this one Today.

tl;dr:
It passed the problematic part and should be able to run normally from this point.

long version:
We've switched to using go 1.16.x by default recently, and that was a jump from 1.14.x IIRC, which caused some changes in the behaviour of the CI, mainly when `go get`ing something.  To fix the issue I've just set `GO111MODULE=auto` to the job and it seems to be happy enough.

But how did we miss this job, but not the other bits of the CI?  The reason is that this job is quite specific and its setup is done directly in Jenkins, which is something we should rethink in the near future.

Anyways, things should be fixed now and sorry for the breakage.
